foreshadow/tests/test_transformers/test_transformers.py

```diff
@@ -571,3 +571,186 @@ def test_sparse_matrix_conversion():
+    [
+        ("StandardScaler", "./foreshadow/tests/test_data/boston_housing.csv"),
+        ("OneHotEncoder", "./foreshadow/tests/test_data/boston_housing.csv"),
+        (
+            "TfidfTransformer",
+            "./foreshadow/tests/test_data/boston_housing.csv",
+        ),
+    ],
+)
+def test_make_pandas_transformer_fit(transformer, input_csv):
+    """Test make_pandas_transformer has initial transformer fit functionality.
+
+        Args:
+            transformer: wrapped transformer
+            input_csv: dataset to test on
+
+    """
+    from importlib import import_module
+    import pandas as pd
+
+    mod = import_module("foreshadow.transformers.externals")
+    transformer = getattr(mod, transformer)()
+    df = pd.read_csv(input_csv)
+    assert transformer.fit(df) == transformer
+
+
+@pytest.mark.parametrize(
+    "transformer,expected_path",
+    [
+        ("StandardScaler", "sklearn.preprocessing"),
+        ("OneHotEncoder", "category_encoders"),
+        ("TfidfTransformer", "sklearn.feature_extraction.text"),
+    ],
+)
+def test_make_pandas_transformer_meta(transformer, expected_path):
+    """Test that the wrapped transformer has proper metadata.
+
+    Args:
+        transformer: wrapped transformer
+        expected_path: path to the initial transformer
+
+    Returns:
+
+    """
+    from importlib import import_module
+
+    mod = import_module(expected_path)
+    expected = getattr(mod, transformer)
+    mod = import_module("foreshadow.transformers.externals")
+    transformer = getattr(mod, transformer)()
+    assert isinstance(transformer, expected)  # should remain a subclass
+    assert type(transformer).__name__ == expected.__name__
+    assert transformer.__doc__ == expected.__doc__
+
+
+@pytest.mark.parametrize(
+    "transformer,kwargs,sk_path,input_csv",
+    [
+        (
+            "StandardScaler",
+            {},
+            "sklearn.preprocessing",
+            "./foreshadow/tests/test_data/boston_housing.csv",
+        ),
+        (
+            "OneHotEncoder",
+            {},
+            "category_encoders",
+            "./foreshadow/tests/test_data/boston_housing.csv",
+        ),
+        (
+            "TfidfTransformer",
+            {},
+            "sklearn.feature_extraction.text",
+            "./foreshadow/tests/test_data/boston_housing.csv",
+        ),
+    ],
+)
+def test_make_pandas_transformer_transform(
+    transformer, kwargs, sk_path, input_csv
+):
+    """Test wrapped transformer has the initial transform functionality.
+
+        Args:
+            transformer: wrapped transformer
+            kwargs: key word arguments for transformer initialization
+            sk_path: path to the module containing the wrapped sklearn
+                transformer
+            input_csv: dataset to test on
+
+    """
+    from importlib import import_module
+    import pandas as pd
+    import numpy as np
+    from scipy.sparse import issparse
+
+    sk_mod = import_module(sk_path)
```

Comment:
This is not absolutely necessary but I saw somewhat similar code between this test and the next few tests, it might be worth looking into pytest set up and tear down. There's an example in test_preprocessor.py

Reply:
@adithyabsk to do